flexget/plugins/sites/newznab.py

```diff
@@ -1,124 +1,561 @@
-            if 'apikey' in config and 'website' in config:
-                params = {
-                    't': config['category'],
-                    'apikey': config['apikey'],
-                    'extended': 1
-                }
-                config['url'] = config['website'] + '/api?' + urlencode(params)
+        if config['search_type'] == 'tv':
+            api_url += '&t=tvsearch'
+        elif config['search_type'] == 'movie':
+            api_url += '&t=movie'
+        else:
+            api_url += '&t=search'
 
-        return config
+        if 'category' in config:
+            categories = config.get('category')
+            if not isinstance(categories, list):
+                categories = [categories]
+            # Convert named categories to its respective categories id number
+            categories = [c if isinstance(c, int) else CATEGORIES[c] for c in categories]
+            if len(categories) > 0:
+                categories_string = ','.join(str(c) for c in categories)
+                api_url += '&cat=%s' % categories_string
 
-    def fill_entries_for_url(self, url, task):
-        entries = []
-        log.verbose('Fetching %s' % url)
+        if config.get('maxage'):
+            api_url += '&maxage=%s' % parse_timedelta(config['maxage']).days
+
+        return api_url
+
+    def convert_to_naive_utc(self, value):
+        parsed_date = None
+        try:
+            parsed_date = dateutil_parse(value, fuzzy=True)
+            try:
+                parsed_date = parsed_date.astimezone(tz.tzutc()).replace(tzinfo=None)
+            except ValueError:
+                parsed_date = parsed_date.replace(tzinfo=None)
+        except ValueError:
+            log.warning('Invalid time format < %s > in Entry < %s >' % value)
+        except Exception as ex:
+            log.trace('Unexpected time field < %s > error: <%s>' % (value, ex))
+
+        return parsed_date
 
+    def convert_to_number(self, valuestring):
+        number = None
         try:
-            r = task.requests.get(url)
-        except task.requests.RequestException as e:
-            log.error("Failed fetching '%s': %s" % (url, e))
+            number = int(valuestring)
+        except ValueError:
+            try:
+                number = float(valuestring)
+            except ValueError:
+                log.trace('Invalid number field < %s >' % valuestring)
+        except Exception as ex:
+            log.trace('Invalid number field < %s > error: <%s>' % (valuestring, ex))
 
-        rss = feedparser.parse(r.content)
-        log.debug("Raw RSS: %s" % rss)
+        return number
 
-        if not len(rss.entries):
-            log.info('No results returned')
+    def string_to_bool(self, value):
+        return value.lower() in ('yes', 'true', 't', '1', 'y')
 
-        for rss_entry in rss.entries:
+    def dump_entry(self, entry):
+        log.verbose('#####################################################################################')
```

Comment:
Yes only for myself, so i can easily spot the start/end of entrys and check if they have all the new fields with good values.